### PR TITLE
Joke

### PR DESCRIPTION
**Pull request by prakritip0**
# Summary of change

* Create folder and files through Node fs
* Read the created files
* Amend the existing files

## Type of change

- [x] New feature

# How Has This Been Tested?

- [x] Manual testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: commands/commands.js [hunk 1]
```diff
@@ -3,31 +3,107 @@ import promptSync from "prompt-sync";
...
+
+
+}
+const readStoredData = () => {
+    const storedData = fs.readFileSync(`${__dirname}/../files/${finalData.name}.json`, "utf-8");
+    const stringifiedStoredData = `${storedData}`;
+    const parsedStoredData = JSON.parse(stringifiedStoredData);
+    // console.log(parsedStoredData);
+
+    console.table([parsedStoredData]);
```
Comment (bisheshbh) [inline commands/commands.js]: Maybe we could do something like this . 
```suggestion
    const storedDataArray = [];
    const storedData = fs.readFileSync(`${__dirname}/../files/${finalData.name}.json`, "utf-8");
    const stringifiedStoredData = `${storedData}`;
    stringifiedStoredData.split("\n").forEach((sd) => {
        console.log(sd)
        storedDataArray.push(JSON.parse(sd))
    })
    console.table(storedDataArray);
```

File: commands/commands.js [hunk 2]
```diff
@@ -3,31 +3,107 @@ import promptSync from "prompt-sync";
...
+            }
+            console.log("\n" + "Your jokes are saved on " + chalk.blue(finalData.name) + chalk.blue(".json") + ". Please enter " + chalk.blue("view") + " to view the file.");
+
+        });
+        console.log("\n" + "Your jokes are saved on " + chalk.blue(finalData.name) + chalk.blue(".json") + ". Please enter " + chalk.blue("view") + " to view the file.");
+    }
+    else {
+        // console.log("exists");
+        // console.log(userDetails);
+        fs.appendFileSync(`${__dirname}/../files/${finalData.name}.json`, userDetails, (err) => {
```
Comment (bisheshbh) [inline commands/commands.js]: Maybe here we could do something like this by adding some line breaks that helps in reading the files later . 
```suggestion
        fs.appendFileSync(`${__dirname}/../files/${finalData.name}.json`, `\n${userDetails}`, (err) => {
```

Comment (prakritip0) [reply commands/commands.js]: Why are we giving a line break before the userDetails and not afterwards?
When we append the file, the newer data gets stores after whatever is already available, no?

Comment (bisheshbh) [reply commands/commands.js]: This is the case for append , so when appending , giving space before adds a space initially making the added line move a line below .

File: commands/commands.js [hunk 3]
```diff
@@ -3,31 +3,113 @@ import promptSync from "prompt-sync";
...
+    const stringifiedStoredData = `${storedData}`;
+    // const parsedStoredData = JSON.parse(stringifiedStoredData);
+    stringifiedStoredData.split("\n").forEach((singleOutputData) => {
+        // console.log(singleOutputData)
+        storedDataArray.push(JSON.parse(singleOutputData));
+    })
+    // console.log(parsedStoredData);
+
+    // console.table([parsedStoredData]);
+    console.log(storedDataArray);
```
Comment (bisheshbh) [inline commands/commands.js:68]: Are we using console log or console.table ?

File: commands/commands.js [hunk 4]
```diff
@@ -3,31 +3,113 @@ import promptSync from "prompt-sync";
...
+                readStoredData();
+                break
+            case "help":
+                showWelcomeBanner();
+                break
             case "x":
+                // console.log("Sad to see you go. Come back soon!!!");
                 return
+            // default:
+            //     showWelcomeBanner();
```
Comment (bisheshbh) [inline commands/commands.js]: Also remove comments before pushing

File: api/joke.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import axios from "axios";
+import chalk from "chalk";
+import { finalData } from "../data/finalData.js";
+
+
+
+export const getJoke = async (category, userName) => {
+    const joke = await axios.get(`https://api.chucknorris.io/jokes/random?category=${category}`);
+    console.log(chalk.green(joke.data.value));
+    finalData.joke = joke.data.value;
```
Comment (bisheshbh) [inline api/joke.js]: You can handle exception here , if in case API got a problem

File: api/category.js [hunk 6]
```diff
@@ -1,15 +1,33 @@
...
     console.log("\n" + chalk.bgGreen("Please enter one of the above categories to hear a joke!"));
+    const chosenCategory = prompt(">");
+    // console.log(Number(chosenCategory));
+    if (isNaN(Number(chosenCategory))) {
+        finalData.category = chosenCategory;
+        return chosenCategory;
+
+    } else {
+        finalData.category = categoryData.data[chosenCategory - 1];
+        return categoryData.data[chosenCategory - 1];
```
Comment (bisheshbh) [inline api/category.js]: As you have wrap the api call inside a try and catch , if the code enters the catch block the categoryData will not have data property , thus categoryData.data might not work ? Maybe you can wrap all the above code also in try block .

File: commands/commands.js [hunk 7]
```diff
@@ -3,31 +3,113 @@ import promptSync from "prompt-sync";
...
+    const fileExistence = fs.existsSync(`${__dirname}/../files`, "utf-8");
+    if (!fileExistence)
+        fs.mkdirSync("files");
+}
+
+
+
+
+
+
```
Comment (bisheshbh) [inline commands/commands.js]: Extra spaces ?

File: index.js [hunk 8]
```diff
@@ -1,16 +1,26 @@
...
+    showAppTitle();
+    showWelcomeBanner();
+    createFileFolder();
     const name = askName();
 
     if (name) {
         console.log(chalk.green(`Hello, ${name}. You can now enter a category to hear a joke.`));
         console.log(chalk.bgWhite("Please press 'c' if you aren't aware of our categories!!"));
     }
     shortcutCommand();
```
Comment (bisheshbh) [inline index.js:18]: What if the name is empty . Maybe we can make the name required , by not letting the program move forward without name ?

File: commands/commands.js [hunk 9]
```diff
@@ -3,31 +3,113 @@ import promptSync from "prompt-sync";
...
+export const showWelcomeBanner = () => {
+    console.log("\n \n" + chalk.bgGreen(constantValues.description) + "\n\n" + constantValues.jokeCategories + chalk.red(shortcuts.see_joke_categories) + "\n" + constantValues.exitProgram + chalk.red(shortcuts.exit_project) + "\n" + constantValues.viewJokesDatabase + chalk.red(shortcuts.view_jokes_database) + "\n" + constantValues.askForHelp + chalk.red(shortcuts.ask_for_help) + "\n");
 }
 
 export const askName = () => {
-    showWelcomeBanner();
     console.log(chalk.green("What is your name?"));
     const name = prompt(">>");
+    finalData.name = name;
```
Comment (bisheshbh) [inline commands/commands.js:34]: Same here , we can handle the case if the name is not provided

File: commands/commands.js (hunk 3 above)
Comment (prakritip0) [reply commands/commands.js:68]: We're using console.log.

**Pull request merged**